### DISCUSSION TITLE
How to power AzureAD settings via configuration?

### DISCUSSION THREAD

**Discussion by willnationsdev** [OrchardCMS/OrchardCore#8432] [Q&A] [answered]
I can't seem to identify the correct `appsettings.json` structure to get what I want.

With my information looking like this...

```json
{
  "AzureADSettings": {
    "DisplayName": <name>,
    "AppId": <appId>,
    "TenantId": <tenantId>,
    "CallbackPath": <path>
  }
}
```

I've tried the following based on the docs:

```json
{
  "OrchardCore": {
    "AzureADSettings": <info>
  }
}

{
  "OrchardCore": {
    "Properties": {
      "AzureADSettings": <info>
    }
  }
}

{
  "OrchardCore": {
    "Default": {
      "AzureADSettings": <info>
    }
  }
}

{
  "OrchardCore": {
    "Default": {
      "Properties": {
        "AzureADSettings": <info>
      }
    }
  }
}
```

Is there something I'm missing? Nothing I put in is overriding the values I entered in the GUI. And I can't clear out the GUI without it giving me validation errors. Am I just doing the config wrong, or should I open a bug report?

Edit: Also tried doing it by replacing `"Default"` with `"OrchardCore.Microsoft.Authentication.AzureAD"` since module names are also applicable. That didn't work either though.

**Comment by jtkech** [comment, marked as answer]
Yes, you would need the `OrchardCore` section, as usualy we use our `IShellConfiguration` that is tenant aware and looks under this section, and you would also need a section whose name is the module name with underscores, here

    OrchardCore_Microsoft_Authentication

But looking at the code, the related code only configures the options from the site settings, so set under the admin UI.

So, you would need some code that configures the related options by using `IShellConfiguration`, this in a module startup that runs after, so that depends on `OrchardCore.Microsoft.Authentication` in its manifest, or by overriding in the startup the `Order` prop with an higher value, the same can be done from the app by using one of our OrchardCoreBuilder helpers.

**Reply by willnationsdev** [reply]
That sounds perfect. Thanks!
